install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -465,8 +465,8 @@ l2announcements:
 l2podAnnouncements:
   # -- Enable L2 pod announcements
   enabled: false
-  # -- Interface used for sending Gratuitous ARP pod announcements
-  interface: "eth0"
+  # -- A regular expression matching interfaces used for sending Gratuitous ARP pod announcements
```

Comment:
Has this feature been in a release? If so, please document this a a breaking change in the upgrade guide

Reply:
Oh, oops. We should keep the `interface` around in the Helm values for the next release. Its going to be deprecated in 1.18 and removed in 1.19, so the breaking change need to occur them. Will fix it, thanks!

Reply:
I also realized the default values may cause an issue. Mainly because we already have a version of the helm chart with this default `eth0` which sucks.

They way I did it now is that we keep the current default, no default for `interfacePattern`. But when the user defined a `interfacePattern` it will replace the `interface` (see configmap), so users do not have to explicitly set `interface` to an empty string if they would like to use the new option.

I think this is the correct way to do it.

Reply:
SGTM. If you want to deprecate the old value in v1.18, we probably should mention that in the upgrade guide still (unless this is done in a different PR)

Reply:
:+1: Updated the upgrade guide as well.